packages/session-manager/src/index.tsx

```diff
@@ -0,0 +1,95 @@
+import { user as userAPI } from '@tryrolljs/api'
+import Client from '@tryrolljs/api-client'
+import { auth } from '@tryrolljs/sdk'
+import {
+  useState,
+  useEffect,
+  useCallback,
+  cloneElement,
+  createContext,
+  PropsWithChildren,
+  ReactElement,
+  useContext,
+} from 'react'
+
+export const SessionContext = createContext<{
+  user?: userAPI.types.GetMeResponseData
+  logIn: () => void
+  logOut: () => Promise<void>
+  error?: unknown
+}>({
+  logIn: () => {},
+  logOut: () => Promise.resolve(),
+})
+
+type Props = PropsWithChildren<{
+  apiClient: Client
+  authSdk: auth.AuthSDK
+}>
+
+const OAUTH_CODE_URL_PARAM_KEY = 'code'
+
+const SessionProvider = ({ apiClient, authSdk, children }: Props) => {
+  const [user, setUser] = useState<userAPI.types.GetMeResponseData>()
+  const [error, setError] = useState<unknown>()
+
+  useEffect(() => {
```

Comment:
is there an advantage to defining these functions within `useEffect` ? or can we define them using `useCallback` ?

Reply:
maybe they all will rely on `user` anyway